### ISSUE TITLE
ErrorException ordering data, if totalRecords=0 an no columnNames were set

### ISSUE THREAD

**Issue by gabrielwelsche** [yajra/laravel-datatables#176]
The stackstrace
```
ErrorException in BaseEngine.php line 242:
Undefined offset: 1
in BaseEngine.php line 242
at HandleExceptions->handleError('8', 'Undefined offset: 1', '/home/gabriel/b2b_next/vendor/yajra/laravel-datatables-oracle/src/yajra/Datatables/Engines/BaseEngine.php', '242', array('column' => '1')) in BaseEngine.php line 242
at BaseEngine->getColumnName('1') in CollectionEngine.php line 96
at CollectionEngine->ordering() in BaseEngine.php line 546
at BaseEngine->orderRecords(false) in BaseEngine.php line 528
at BaseEngine->make(true, true) in CollectionEngine.php line 194
at CollectionEngine->make(true) in QuoteRequestController.php line 143
```

Bugfix e.g. in yajra\Datatables\Engines\BaseEngine::make()
```php
public function make($mDataSupport = false, $orderFirst = false)
{
    $this->totalRecords = $this->count();
    if ($this->totalRecords) {
      $this->orderRecords( ! $orderFirst);
      $this->filterRecords();
      $this->orderRecords($orderFirst);
    }
    $this->paginate();

    return $this->render($mDataSupport);
}
```

**Comment by yajra** [comment]
@gabrielwelsche, thanks for the inputs. What version are you using? I tried testing using empty records and did not encountered the same issue.

**Comment by gabrielwelsche** [comment]
composer: "yajra/laravel-datatables-oracle": "~5.0",

.../vendor/yajra/laravel-database-oracle/change-log.md
```
[...]
##Change Log

###v5.11.7
    - Fix escaping of row when using eager loading. Fix #164
[...]
```

I think I use the latest release:

Here is the config setting, which produce the error: 
```
$('#table').DataTables({
   [...]
     sort: true,
     searching: true,
     stateSave: true,   // remember "sort" + page
     serverSide: true,
    columns:{
      {data: 'id'},
      {data: 'article_id'},
      [...]
    }
}); 
```

When I change my "columns", then it works properly:
```
$('#table').DataTables({
   [...]
    columns:{
      {data: 'id', name: 'id'},
      {data: 'article_id', name: 'article_id'},
      [...]
    }
}); 
```

Thank you very much for your work!

Gabriel

**Comment by yajra** [comment]
@gabrielwelsche , You MUST use the second configuration when using `make(true)` because the `name` attribute is being used as the field name when searching/sorting the column.

**Comment by yajra** [comment]
@gabrielwelsche, I just reviewed your suggestion and implemented it. Released on v5.11.8. Thanks!

**Comment by gabrielwelsche** [comment]
Thank you very much!!!!
